### PR TITLE
feat(shell): Add panel-top slot

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #6389 

## Summary
Adds a slot for positioning content at the top of the center area of a Shell. As part of the larger set of layout updates (#6449), this will eventually be a use case for Shell Panel.

For now, this will continue to use the Shell Center Row, just like the `center-row` slot. That slot has been "deprecated" (as much as you can deprecate a slot), and a new `panel-bottom` slot has been added.

This has some of the same "swim lane" configuration corners as the current Shell Panel when combining `detached` without `content-behind`, but those exist already and will be alleviated with documentation and further component enhancements.

Screen Shot 2023-04-04 at 12 52 16 PM
Screen Shot 2023-04-04 at 12 51 40 PM
Screen Shot 2023-04-04 at 12 51 30 PM
Screen Shot 2023-04-04 at 12 52 29 PM
Screen Shot 2023-04-04 at 1 21 06 PM
Screen Shot 2023-04-04 at 1 20 43 PM

cc @driskull @asangma

### CODE REVIEW

File: src/components/shell/shell.tsx [hunk 1]
```diff
@@ -13,7 +13,9 @@ import { CSS, SLOTS } from "./resources";
  * @slot footer - A slot for adding footer content. This content will be positioned at the bottom of the component.
  * @slot panel-start - A slot for adding the starting `calcite-shell-panel`.
  * @slot panel-end - A slot for adding the ending `calcite-shell-panel`.
- * @slot center-row - A slot for adding content to the center row.
+ * @slot panel-top - A slot for adding the top `calcite-shell-panel`.
+ * @slot panel-bottom - A slot for adding the bottom `calcite-shell-panel`.
+ * @slot center-row - A slot for adding the bottom `calcite-shell-panel`. (Deprecated - use `panel-bottom` instead)
```
Comment (driskull) [inline src/components/shell/shell.tsx:18]: I think this should start with deprecated for consistency.

```
 * @slot center-row - [Deprecated] use "panel-bottom" instead. A slot for adding content to the center row.
```

**Review by driskull — COMMENTED** [review]
Looks good! Needs a screenshot and e2e test though.

**Comment by macandcheese** [comment]
@ashetland @SkyeSeitz - I updated the helper that generates story content for Shell Panel, Shell Center Row, so there may be some additional changes in Chromatic in addition to the ones added.

File: src/components/shell/shell.tsx (hunk 1 above)
Comment (macandcheese) [reply src/components/shell/shell.tsx]: Updated - @geospatialem this is the correct way to format "deprecated" slots right? We don't have any other current cases. Would this be a breaking change even though it's a "doc only" deprecation?

Comment (geospatialem) [reply src/components/shell/shell.tsx]: We can verify in the next doc update with `1.3.0`, but it seems to be on par with what we had prior to 1.0. 👍🏻 

It looks like any text with `[deprecated]` mentioned receives the deprecation chip on the docs site (lower/upper case character combinations shouldn't matter). 

So `[Deprecated]` seems like a good pattern, and is what we had as of late November 2022 in [shell.txt](https://github.com/Esri/calcite-components/pull/5819/files#diff-b5b8c066968e2720736cedb2d71b8f55fa99bae5fbb8d8b28729cb47212de564) with `[DEPRECATED]` - so pick your poison with lower/upper case combinations.

File: src/components/shell/shell.tsx [hunk 2]
```diff
@@ -13,9 +13,9 @@ import { CSS, SLOTS } from "./resources";
  * @slot footer - A slot for adding footer content. This content will be positioned at the bottom of the component.
  * @slot panel-start - A slot for adding the starting `calcite-shell-panel`.
  * @slot panel-end - A slot for adding the ending `calcite-shell-panel`.
- * @slot panel-top - A slot for adding the top `calcite-shell-panel`.
- * @slot panel-bottom - A slot for adding the bottom `calcite-shell-panel`.
- * @slot center-row - A slot for adding the bottom `calcite-shell-panel`. (Deprecated - use `panel-bottom` instead)
+ * @slot panel-top - A slot for adding the top `calcite-shell-center-row`.
+ * @slot panel-bottom - A slot for adding the bottom `calcite-shell-center-row`.
+ * @slot center-row - [Deprecated] use "panel-bottom" instead. A slot for adding the bottom `calcite-shell-center-row`.
```
Comment (geospatialem) [inline src/components/shell/shell.tsx:18]: nitpick: Add backticks for API doc consistency to the referenced slot name.

```
[Deprecated] use `"panel-bottom"` instead.
```

**Comment by macandcheese** [comment]
Merging following Chromatic approval: https://www.chromatic.com/pullrequest?appId=6266d45509d7eb004aa254fb&number=6730

**Pull request merged**